tools/database/gen_db/main.go

```diff
@@ -171,7 +171,19 @@ func main() {
 	}
 
 	for _, item := range atlaslootDB.Items {
-		if _, ok := db.Items[item.Id]; ok {
+		if src, ok := db.Items[item.Id]; ok {
+
+			// Manually filter sources from AtlasLoot to prevent duplicates
+			item.Sources = core.FilterSlice(item.Sources, func(source *proto.UIItemSource) bool {
+				if rep := source.GetRep(); rep != nil {
+					for _, srcSource := range src.Sources {
+						if srcRep := srcSource.GetRep(); srcRep != nil {
+							return srcRep.FactionId != rep.FactionId
+						}
+					}
+				}
+				return true
+			})
```

Comment:
The following is equivalent to your code and easier to understand:
```
firstDBRepFaction := proto.Faction(-1)

for _, dbItemSource := range dbItem.Sources {
    if dbRepSource := dbItemSource.GetRep(); dbRepSource != nil {
        firstDBRepFaction = dbRepSource.FactionId
        break
    }
}

atlaslootItem.Sources = core.FilterSlice(atlaslootItem.Sources, func(atlaslootItemSource *proto.UIItemSource) bool {
    atlaslootRepSource := atlaslootItemSource.GetRep()
    return (atlaslootRepSource == nil) || (atlaslootRepSource.FactionId != firstDBRepFaction)
})
```

Reply:
Also, are you sure you mean to compare the `FactionId` field rather than the `RepFactionId` field?